src/studio/src/designer/backend.Tests/Factories/ModelFactory/XsdToCSharpTests.cs

```diff
@@ -0,0 +1,131 @@
+using System;
+using System.IO;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+using System.Xml.Serialization;
+using Altinn.Studio.Designer.Factories.ModelFactory;
+using Altinn.Studio.Designer.ModelMetadatalModels;
+using Designer.Tests.Utils;
+using Manatee.Json.Schema;
+using Xunit;
+
+namespace Designer.Tests.Factories.ModelFactory
+{
+    public class XsdToCSharpTests
+    {
+        [Fact]
+        public async void ConvertXsdToJsonSchema_CorrectXMLParsed()
+        {
+            // Arrange
+            using XmlReader xsdReader = XmlReader.Create(LoadTestData("Model/Xsd/Brønnøysundregistrene_ReelleRettighetshavere_M_2021-11-22_6900_46864_SERES.xsd"));
+            XsdToJsonSchema target = new XsdToJsonSchema(xsdReader);
+
+            // Act
+            JsonSchema xsdasJsonSchema = target.AsJsonSchema();
+
+            JsonSchemaToInstanceModelGenerator converter = new JsonSchemaToInstanceModelGenerator("test", "test", xsdasJsonSchema);
+            ModelMetadata modelMetadata = converter.GetModelMetadata();
+
+            string classes = GenerateCSharpClasses(modelMetadata);
+            Assembly assembly = Compiler.CompileToAssembly(classes);
+            Type type = assembly.GetType("Altinn.App.Models.ReelleRettighetshavere_M");
+            var modelInstance = assembly.CreateInstance(type.FullName);
+
+            Stream xmlStream = TestDataHelper.LoadDataFromEmbeddedResource("Designer.Tests._TestData.ModelData.Brønnøysundregistrene_ReelleRettighetshavere_M_2021-11-22_6900_46864_SERES.CorrectOrder.xml");
+
+            var xmlObject = await DeserializeXmlAsync(xmlStream, type);
+
+            XmlSerializer serializer = new XmlSerializer(type);
+            using MemoryStream stream = new MemoryStream();
+            serializer.Serialize(stream, xmlObject);
+            stream.Position = 0;
+
+            StreamReader reader = new StreamReader(stream);
+            string text = reader.ReadToEnd();
+
+            xmlStream = TestDataHelper.LoadDataFromEmbeddedResource("Designer.Tests._TestData.ModelData.Brønnøysundregistrene_ReelleRettighetshavere_M_2021-11-22_6900_46864_SERES.CorrectOrder.xml");
```

Comment:
Can we avoid reading the same resource twice?

Reply:
Tried but always ended up with the stream being closed